### PR TITLE
Adding HIP/CUDA jobs in CI

### PR DESCRIPTION
**Pull request by kab163**
For a while, we have made hip+openmp jobs a conflict within the spack. I don't think that restriction holds anymore. This ticket will add a hip+openmp job, but keep the conflict for hip+openmp_target.

Along the same lines, we have disabled cuda+shared jobs because there was a restriction at one point between cuda jobs and building with shared libs. That no longer applies, so we will also add '+shared' to an existing cuda job to test.

### CODE REVIEW

**Comment by adrienbernede** [comment]
@kab163 I am currently working on revamp the CI. The change here is pretty minimal, so I don’t think I need to worry too much about having this merged in before I start.

File: .gitlab/corona-jobs.yml [hunk 1]
```diff
@@ -11,5 +11,5 @@ rocm_4_5_2_fortran_clang_13_0_0 (build and test on corona):
 
 rocm_4_5_2_fortran_clang_13_0_0(build and test on corona):
```
Comment (adrienbernede) [inline .gitlab/corona-jobs.yml]: I think it will be wise to rename one of those job so that they are easier to differentiate.

Comment (kab163) [reply .gitlab/corona-jobs.yml]: I was thinking the same thing!

**Review by davidbeckingsale — APPROVED** [review]
👍

**Review by mcfadden8 — APPROVED** [review]

**Comment by adrienbernede** [comment]
Oops, so this was merged. I’ll need to adapt #723

**Comment by adrienbernede** [comment]
Done.

**Pull request merged**